### PR TITLE
Upgrade Spire to use CSI driver

### LINKED ISSUE #130: Update SSF to use spiffe-csi
Possible Solution or Alternative
---------------------------------
Update our spire deployment to use the `spiffe-csi`

Context
-------
Likely a better option in terms of security as it limits the need for the `hostpath` to share the workload API. Only the spire-agent would be using the `hostpath` and not the other workload pods.

### PR DESCRIPTION
**Pull request by sudo-bmitch**
This bumps up the Spire helm chart for the latest release, and the helm chart also adds support for the CSI driver so that workloads no longer need to perform a hostPath mount. The chains patch was updated to use the driver.

Fixes #130, replaces #354

Signed-off-by: Brandon Mitchell <git@bmitch.net>

### CODE REVIEW

**Comment by sudo-bmitch** [comment]
cc @developer-guy and @dentrax

File: platform/vendor/spire/chart/templates/spiffe-csi-driver.yaml
Comment (developer-guy) [inline platform/vendor/spire/chart/templates/spiffe-csi-driver.yaml:1]: maybe we can add the same condition over here:

```yaml
{{- if .Values.csi.enabled }}
...
{{- end }}
```

**Review by developer-guy — APPROVED** [review]
not a maintainer but, it looks perfect, lgtm!

**Comment by sudo-bmitch** [comment]
CI will be fixed by #356

File: platform/vendor/spire/chart/templates/spiffe-csi-driver.yaml
Comment (sudo-bmitch) [reply platform/vendor/spire/chart/templates/spiffe-csi-driver.yaml:2]: Good catch, done!

**Review by bradbeck — APPROVED** [review]

**Comment by developer-guy** [comment]
Everything seems ready for merge, is there anything left to do before merging it? @sudo-bmitch

File: platform/components/spire/values.yaml
Comment (pxp928) [inline platform/components/spire/values.yaml:8]: This will require changes to Tekton Chains to use the CSI driver. I assume it will be a separate PR?

Comment (sudo-bmitch) [reply platform/components/spire/values.yaml:8]: There's an update to patch_spire.json in this PR that changes `hostPath` to `csi`.

Comment (pxp928) [reply platform/components/spire/values.yaml:8]: missed that. Thank you!

File: platform/vendor/spire/chart/templates/server-cluster-role.yaml
Comment (sudo-bmitch) [reply platform/vendor/spire/chart/templates/server-cluster-role.yaml:16]: @developer-guy is this needed? If not, I'm going to remove it from upstream and revendor.

Comment (developer-guy) [reply platform/vendor/spire/chart/templates/server-cluster-role.yaml]: we noticed an error when we upgraded the images to version v1.4.4,

Comment (developer-guy) [reply platform/vendor/spire/chart/templates/server-cluster-role.yaml]: it might be related to _k8s-workload-registrar_ component, though 🤔

**Review by bradbeck — APPROVED** [review]

File: platform/vendor/spire/chart/templates/server-cluster-role.yaml
Comment (developer-guy) [reply platform/vendor/spire/chart/templates/server-cluster-role.yaml]: _https://github.com/spiffe/spire/commit/cc259530565bc14c56f1da531d8403115c798b28_

@sudo-bmitch ☝️

Comment (developer-guy) [reply platform/vendor/spire/chart/templates/server-cluster-role.yaml]: This is related with the following issue 👇

_https://github.com/buildsec/frsca/issues/353_

Comment (sudo-bmitch) [reply platform/vendor/spire/chart/templates/server-cluster-role.yaml]: Got it. Let's look at that separately for the workload-registrar. If it's needed in the helm chart, we can put it behind a value flag.

**Pull request merged**
